pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ProtoParserTools.java

```diff
@@ -216,8 +217,11 @@ public static double readDouble(final ReadableSequentialData input) throws IOExc
      */
     public static String readString(final ReadableSequentialData input) throws IOException {
         final int length = input.readVarInt(false);
+        if (input.remaining() < length) {
+            throw new BufferOverflowException();
+        }
         byte[] bytes = new byte[length];
-        input.readBytes(bytes);
+        final var read = input.readBytes(bytes);
```

Comment:
We need to check the read here, it should match length otherwise we need to throw exception.

Reply:
Actually, good catch. I had it implemented that way and decided later to do the check before reading. This way it will fail faster - and potentially work faster (same as all the rest implementations are done in BufferedData. It is faster too, to not have to read at all and just fail.
The check is done in the pre-read if.